### PR TITLE
Divyanshu | Test | Implementation of Zimbabwe VAT in GIDDH

### PR DESCRIPTION
**Pull request by dvCodeWorld**
Zimbabwe - https://app.clickup.com/t/86cuy39kd
Kenya - https://app.clickup.com/t/86cv1ctb7

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 1]
```diff
@@ -43,32 +43,39 @@
...
-                                    </ng-container>
-                                    <ng-container *ngIf="!connectToHMRCUrl">
-                                        {{ localeData?.connected_to_hmrc }}
-                                    </ng-container>
 
-                                </a>
+                                    <div class="ml-auto d-flex justify-content-center column-gap1 align-items-center">
+                                        <div *ngIf="isZimbabweCompany" class="currency-width">
+                                            <dropdown-field [label]="commonLocaleData?.app_currency" [options]="vatReportCurrencyList"
+                                                (selectedOption)="vatReportCurrency = $event?.value; currentCurrencySymbol = $event.additional.symbol"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                (selectedOption)="vatReportCurrency = $event?.value; currentCurrencySymbol = $event?.additional?.symbol"
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 2]
```diff
@@ -43,32 +43,39 @@
...
+                                                [placeholder]="commonLocaleData?.app_select" [defaultValue]="vatReportCurrencyList[0]?.label"
+                                                [required]="true"></dropdown-field>
+                                        </div>
+                                        <!-- Adsofjsopdfdfds nnnfdsfndskfsdfd   Add fake condition || true -->
+                                        <div *ngIf="taxNumber && !isUKCompany && (isZimbabweCompany || isKenyaCompany) || true" class="export-btn">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: fix condition

File: apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+export class WithHeldSettingComponent implements OnInit, OnDestroy {
+    /** Observable to unsubscribe all the store listeners to avoid memory leaks */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** This will hold local JSON data */
+    public localeData: any = {};
+    /** This will hold common JSON data */
+    public commonLocaleData: any = {};
+    /** This will hold the value out/in to open/close setting sidebar popup */
+    public asideGstSidebarMenuState: string = 'in';
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts]:
```suggestion
    public asideTaxSidebarMenuState: string = 'in';
```

File: apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    * Lifecycle hook for initialization
+    *
+    * @memberof WithHeldSettingComponent
+    */
+    public ngOnInit(): void {
+        document.querySelector('body').classList.add('gst-sidebar-open');
+    }
+
+    public changeTaxPercentage(): void {
+        console.log(this.taxPercentage.value);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts]: remove console and method if no use

File: apps/web-giddh/src/app/voucher/voucher.component.ts [hunk 5]
```diff
@@ -6167,11 +6169,20 @@ export class VoucherComponent implements OnInit, OnDestroy, AfterViewInit, OnCha
...
                 this.showVATNo = false;
                 this.getOnboardingForm('AE')
-            } else if (name === 'United Kingdom') {
+            } else if (this.vatSupportedCountries.includes(name)) {
                 this.showGSTINNo = false;
                 this.showTRNNo = false;
                 this.showVATNo = true;
-                this.getOnboardingForm('GB')
+
+                let countryCode = null
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/voucher/voucher.component.ts]:
```suggestion
                let countryCode = null;
```

File: apps/web-giddh/src/app/voucher/voucher.component.ts [hunk 6]
```diff
@@ -6167,11 +6169,20 @@ export class VoucherComponent implements OnInit, OnDestroy, AfterViewInit, OnCha
...
+                if (name === 'United Kingdom') {
+                    countryCode = 'GB';
+                }else if (name === 'Zimbabwe') {
+                    countryCode = 'ZW';
+                }else if (name === 'Kenya') {
+                    countryCode = 'KE';
+                }
+                this.getOnboardingForm(countryCode)
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/voucher/voucher.component.ts]:
```suggestion
                this.getOnboardingForm(countryCode);
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/with-held-setting/with-held-setting.component.ts]: Done

File: apps/web-giddh/src/app/voucher/voucher.component.ts (hunk 5 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/voucher/voucher.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 7]
```diff
@@ -43,32 +43,38 @@
...
 
-                                </a>
+                                    <div class="ml-auto d-flex justify-content-center column-gap1 align-items-center">
+                                        <div *ngIf="isZimbabweCompany" class="currency-width">
+                                            <dropdown-field [label]="commonLocaleData?.app_currency" [options]="vatReportCurrencyList"
+                                                (selectedOption)="vatReportCurrency = $event?.value; currentCurrencySymbol = $event?.additional?.symbol"
+                                                [placeholder]="commonLocaleData?.app_select" [defaultValue]="vatReportCurrencyList[0]?.label"
+                                                [required]="true"></dropdown-field>
+                                        </div>
+                                        <div *ngIf="taxNumber && !isUKCompany && (isZimbabweCompany || isKenyaCompany)" class="export-btn">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                        <div *ngIf="taxNumber && (isZimbabweCompany || isKenyaCompany)" class="export-btn">
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 7 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**